index.js

```diff
@@ -292,27 +304,27 @@ module.exports = class Filter extends Plugin {
     });
   }
 
-  async _handleFile(relativePath, srcDir, destDir, entry, outputPath, forceInvalidation, isChange, stats) {
+  async _handleFile(relativePath, entry, outputPath, forceInvalidation, isChange, stats) {
     stats.handleFile++;
 
     let handleFileStart = process.hrtime();
+    let destDir = this.outputPath + '/' + outputPath;
     try {
       let result;
-      let srcPath = srcDir + '/' + relativePath;
 
       if (this.canProcessFile(relativePath, entry)) {
         stats.processed++;
         if (this._outputLinks[outputPath] === true) {
           delete this._outputLinks[outputPath];
-          fs.unlinkSync(outputPath);
+          fs.unlinkSync(destDir);
         }
-        result = await this.processAndCacheFile(srcDir, destDir, entry, forceInvalidation, isChange, stats);
+        result = await this.processAndCacheFile(relativePath, entry, forceInvalidation, isChange, stats);
       } else {
         stats.linked++;
         if (isChange) {
-          fs.unlinkSync(outputPath);
+          fs.unlinkSync(destDir);
```

Comment:
this is breaking encapsulation.

Reply:
We are yet to add the `unlinkSync` to `broccoli-plugin`. 
PRs: 
https://github.com/broccolijs/broccoli-plugin/pull/46
https://github.com/broccolijs/broccoli-plugin/pull/47